doc/sphinx/proofs/automatic-tactics/auto.rst

```diff
@@ -722,11 +784,12 @@ Creating Hints
 
 .. cmd:: Print HintDb @ident
 
-   This command displays all hints from database :n:`@ident`.  Hints
-   in each group ("For ... ->") are shown in the order in which they will be tried
-   (first to last).  The groups are shown ordered alphabetically on the last component of
-   the symbol name.  Note that hints with the same cost are tried in
-   reverse of the order they're defined in, i.e., last to first.
+   This command displays all hints from database :n:`@ident`.  Hints are grouped by
+   the :term:`head constants <head constant>` of their patterns ("For ... ->").
+   The groups are shown ordered alphabetically on the last component of the head
+   constant name.   Within each group, hints are shown in the order in which they
+   will be tried (first to last).  Note that hints with the same cost are tried in
+   reverse of the order they're defined in, i.e., last defined is used first.
```

Comment:
Could we mention hint modes here? I think they are also part of the output.

Reply:
Not yet.  I need to spend time playing with modes before I can say anything intelligent.